src/api/getBridgedBlocks.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+import { ApiPromise } from '@polkadot/api';
+import { VoidFn } from '@polkadot/api/types';
+import { useGetApi, ReturnApi } from './useGetApi';
+
+interface DataType {
+  api: ApiPromise;
+  apiMethod: string;
+  separator: string;
+  setter: any;
+  arg1?: any;
+}
+
+export const getBridgedBlocks = async ({ api, apiMethod, separator, setter, arg1 }: DataType): Promise<VoidFn> => {
```

Comment:
Why couldn't this be two separate methods with proper typing and argument names? `arg1` seems a bit like an overkill.

Reply:
Im trying to figure out if there is a way of a reusable hook/function that will bind Api usage with ApiSubscriptions together. I want to be reusable but not too generic that will make typings disappear (that would be a NO)

Reply:
what about if you create a third hook called `useApiCallSubscribed` or `useApiCallWithSubscription` that wraps the hooks `useGetApi` and `useApiSubscription`?

```
function useApiCallSubscribed(getBlock, ...args){
   const {executeCb, data, error, isLoading} = useGetApi(getBlock, ...args):
   useApiSubscription(executeCb, isReady)

   return {data, error, isLoading}
}
```

Reply:
IMHO i think that having two separate files makes the things easier to understand since you wouldn't need a separator either an `if` block logic.

Reply:
> what about if you create a third hook called `useApiCallSubscribed` or `useApiCallWithSubscription` that wraps the hooks `useGetApi` and `useApiSubscription`?
> 
> ```
> function useApiCallSubscribed(getBlock, ...args){
>    const {executeCb, data, error, isLoading} = useGetApi(getBlock, ...args):
>    useApiSubscription(executeCb, isReady)
> 
>    return {data, error, isLoading}
> }
> ```

I have already created that since yesterday (unfortunately for some reasons there are some errors found (even though I cannot find a leak)). I will push the code anyway so you can see what I mean.

Reply:
[This](https://github.com/paritytech/parity-bridges-ui/blob/nik-fix-apis-on-root/src/api/useApiCallAndSubscriptions.ts) is the newly created one that takes care of everything. Again though - still draft and still working on it

Reply:
@tomusdrw if function is separated into 2, then there is no way to "automate" the useGetBridges hook in order to be called from useEffects etc. This actually breaks the concept introduced in `idea.ts`. Of course it can be done but it is a different approach